### PR TITLE
Testing all WinForms controls have unit tests coverage

### LINKED ISSUE #10453: Ensure that all WinForms controls have unit tests coverage for public APIs
- [x]  Investigate if we can re-enable code coverage on internal builds
        We enabled it in #514, however Coverage option is disabled in the build pipeline -  https://github.com/dotnet/winforms/blob/95e83ed6e86c93c10020a7c92a6c9d459f65f6a5/eng/pipelines/build.yml#L33.
- [x] Generate list of controls with missing test coverage for public methods
- [ ] Work with @Olina-Zhang's team on adding the missing scenarios

Following controls are missing test coverage for public methods:
**System.Windows.Forms**
- [x] Button https://github.com/dotnet/winforms/pull/10693
- [x] CheckBox https://github.com/dotnet/winforms/pull/10693, https://github.com/dotnet/winforms/pull/11249
- [x] FlatButton https://github.com/dotnet/winforms/pull/10693, https://github.com/dotnet/winforms/pull/12882
- [x] RadioButton https://github.com/dotnet/winforms/pull/10693, https://github.com/dotnet/winforms/pull/11687
- [x] ComboBox https://github.com/dotnet/winforms/pull/11381, https://github.com/dotnet/winforms/pull/11533, https://github.com/dotnet/winforms/pull/11565, https://github.com/dotnet/winforms/pull/11699, https://github.com/dotnet/winforms/pull/12038
- [x] DataGridView https://github.com/dotnet/winforms/pull/11541, https://github.com/dotnet/winforms/pull/11556, https://github.com/dotnet/winforms/pull/11618, https://github.com/dotnet/winforms/pull/11691, https://github.com/dotnet/winforms/pull/12622, https://github.com/dotnet/winforms/pull/12814, https://github.com/dotnet/winforms/pull/12768, https://github.com/dotnet/winforms/pull/12766, https://github.com/dotnet/winforms/pull/12673
- [x] DateTimePicker https://github.com/dotnet/winforms/pull/11033, https://github.com/dotnet/winforms/pull/11106, https://github.com/dotnet/winforms/pull/11318, https://github.com/dotnet/winforms/pull/11385, https://github.com/dotnet/winforms/pull/11534, https://github.com/dotnet/winforms/pull/11463
- [x] GroupBox https://github.com/dotnet/winforms/pull/11688
- [x] Label  https://github.com/dotnet/winforms/pull/10787, https://github.com/dotnet/winforms/pull/10976, https://github.com/dotnet/winforms/pull/11521
- [x] LinkLabel https://github.com/dotnet/winforms/pull/12572
- [x] CheckedListBox https://github.com/dotnet/winforms/pull/11265, https://github.com/dotnet/winforms/pull/11317, 
https://github.com/dotnet/winforms/pull/11384, https://github.com/dotnet/winforms/pull/11647
- [x] ListBox https://github.com/dotnet/winforms/pull/11624, https://github.com/dotnet/winforms/pull/11724，https://github.com/dotnet/winforms/pull/12952，https://github.com/dotnet/winforms/pull/12907，
- [x] ListView https://github.com/dotnet/winforms/pull/11264, https://github.com/dotnet/winforms/pull/11380, https://github.com/dotnet/winforms/pull/11701
- [x] MenuStrip #2943
- [x] MonthCalendar https://github.com/dotnet/winforms/pull/11149, https://github.com/dotnet/winforms/pull/11217, https://github.com/dotnet/winforms/pull/11519, https://github.com/dotnet/winforms/pull/11619
- [x] ProgressBar https://github.com/dotnet/winforms/pull/11074
- [x] PropertyGrid https://github.com/dotnet/winforms/pull/11476, https://github.com/dotnet/winforms/pull/11587, https://github.com/dotnet/winforms/pull/11711
- [x] RichTextBox https://github.com/dotnet/winforms/pull/10994
https://github.com/dotnet/winforms/pull/10921
https://github.com/dotnet/winforms/pull/11034
- [x] MaskedTextBox https://github.com/dotnet/winforms/pull/10978, https://github.com/dotnet/winforms/pull/11231, https://github.com/dotnet/winforms/pull/11286, https://github.com/dotnet/winforms/pull/11376
- [x] TextBox https://github.com/dotnet/winforms/pull/11021
- [x] ToolStrip https://github.com/dotnet/winforms/pull/11464, https://github.com/dotnet/winforms/pull/11848, https://github.com/dotnet/winforms/pull/11773, https://github.com/dotnet/winforms/pull/11848, https://github.com/dotnet/winforms/pull/11887, https://github.com/dotnet/winforms/pull/12111, https://github.com/dotnet/winforms/pull/12082, https://github.com/dotnet/winforms/pull/12043, https://github.com/dotnet/winforms/pull/11999, https://github.com/dotnet/winforms/pull/11887, https://github.com/dotnet/winforms/pull/12238, https://github.com/dotnet/winforms/pull/12166, https://github.com/dotnet/winforms/pull/12347, https://github.com/dotnet/winforms/pull/12316, https://github.com/dotnet/winforms/pull/12394, https://github.com/dotnet/winforms/pull/12736
- [x] TrackBar https://github.com/dotnet/winforms/pull/11066, https://github.com/dotnet/winforms/pull/11397, https://github.com/dotnet/winforms/pull/12427
- [x] TreeView https://github.com/dotnet/winforms/pull/11293, https://github.com/dotnet/winforms/pull/11361, https://github.com/dotnet/winforms/pull/11686，https://github.com/dotnet/winforms/pull/12908
- [x] DomainUpDown https://github.com/dotnet/winforms/pull/11331
- [x] WebBrowser https://github.com/dotnet/winforms/pull/10755
- [x] NumericUpDown https://github.com/dotnet/winforms/pull/11522

### PR DESCRIPTION
**Pull request by SimonZhao888**
Fixes #10453

## Proposed changes

- Enable code coverage on public PR builds.

## Customer Impact

- No

## Regression? 

- No

## Risk

-Yes

## Screenshots 

image
image

## Test methodology 

- Pipeline link [Testing all WinForms controls have unit tests coverage](https://dev.azure.com/dnceng-public/public/_build/results?buildId=509837&view=codecoverage-tab)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/10476)

### CODE REVIEW

**Comment by Tanya-Solyanik** [comment]
@sharwell  - do you have any recommendations on how to re-enable code coverage, or sample repos we could take a look at?

**Comment by RussKie** [comment]
I've done this in a few repo using https://github.com/microsoft/codecoverage/. E.g., https://github.com/dotnet/aspire/pull/569.
Please let ping me if you have any questions or need help.

**Comment by sharwell** [comment]
My recommendation would be to follow https://github.com/dotnet/roslyn-analyzers/, using coverlet, ReportGenerator, and codecov.io. At this time I cannot recommend the use of microsoft/codecoverage due to its licensing policy and lack of benefits over the more permissive/open/well-established packages which I recommend instead.

**Comment by RussKie** [comment]
> My recommendation would be to follow [dotnet/roslyn-analyzers](https://github.com/dotnet/roslyn-analyzers/), using coverlet, ReportGenerator, and codecov.io. At this time I cannot recommend the use of microsoft/codecoverage due to its licensing policy and lack of benefits over the more permissive/open/well-established packages which I recommend instead.

@sharwell why are you suggesting to use 3P service (i.e., codecov.io) over using the MS services? This implies sending data to 3P.
Also, what is the problem with microsoft/codecoverage licensing? 

/cc: @jakubch1

**Comment by sharwell** [comment]
> why are you suggesting to use 3P service (i.e., codecov.io) over using the MS services?

I haven't seen any 1P service that works in this space. Specifically, all 1P services that I've observed focus on the code coverage for the repository as a whole, a largely irrelevant feature. The single highest-value code coverage feature which any system _must_ implement to even be considered viable is allowing a pull request reviewer to see the code coverage impact of that pull request. This includes the following two items:

1. (Most important) Show the code coverage of the lines of code changed by that pull request
2. (Secondary) Show the changes to code coverage occurring on lines of code not touched by that pull request

> Also, what is the problem with microsoft/codecoverage licensing?

It's not open source, but there are other open-source alternatives which have been around longer and perform just as well. I can't condone choosing a closed ecosystem without clear benefits.

File: eng/MergeCoverageData.ps1 [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  [string][Alias('c')]$configuration = "Debug"
+)
+
+. $PSScriptRoot/common/tools.ps1
+$reportgeneratorDir = Join-Path $ToolsDir "reportgenerator"
+$reportgeneratorPath = Join-Path $reportgeneratorDir "tools\net6.0\any\ReportGenerator.exe"
+if (-not (Test-Path $reportgeneratorPath)) {
+    Invoke-WebRequest -Uri https://www.nuget.org/api/v2/package/dotnet-reportgenerator-globaltool/5.1.10 -UseBasicParsing -OutFile $TempDir\reportgenerator.zip | Out-Null
+    Expand-Archive -Path $TempDir\reportgenerator.zip $reportgeneratorDir
+}
```
Comment (RussKie) [inline eng/MergeCoverageData.ps1:17]: This is all unnecessary, the report generator has a .NET Tool version, which is installable via [.config/dotnet-tools.json](https://github.com/dotnet/aspire/pull/569/files#diff-7afd3bcf0d0c06d6f87c451ef06b321beef770ece4299b3230bb280470ada2f6) by the Arcade.

The generator is invocable via:
```
./.dotnet/dotnet reportgenerator <args>
```

File: src/Microsoft.VisualBasic.Forms/tests/UnitTests/Microsoft.VisualBasic.Forms.Tests.vbproj [hunk 2]
```diff
@@ -11,6 +11,7 @@
 
   <ItemGroup>
     <PackageReference Include="xunit.stafact" Version="$(XUnitStaFactPackageVersion)" />
+    <PackageReference Include="coverlet.msbuild" Version="$(CoverletMSBuildPackageVersion)" />
```
Comment (RussKie) [inline src/Microsoft.VisualBasic.Forms/tests/UnitTests/Microsoft.VisualBasic.Forms.Tests.vbproj:14]: Perhaps a more strategic approach would be to move all test-related definitions into own Tests.props and Tests.targets. 
```diff
diff --git a/Directory.Build.props b/Directory.Build.props
index b84b51cf5..c7bc1bb8f 100644
--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -4,6 +4,7 @@
   <Import Project="$(RepositoryEngineeringDir)CodeStyle.props" />
   <Import Project="$(RepositoryEngineeringDir)FacadeAssemblies.props" />
   <Import Project="$(RepositoryEngineeringDir)ApiCompatibility\PublicApiAnalyzer.props" />
+  <Import Project="$(RepositoryEngineeringDir)Tests.props" Condition="'$(IsTestProject)' == 'true'" />
 
   <PropertyGroup>
     <!-- Remove hardcoded .NET versions when Arcade is upgraded to .NET 9.0. https://github.com/dotnet/winforms/issues/9851-->
```

```diff
diff --git a/Directory.Build.targets b/Directory.Build.targets
index abd672428..bec91d07c 100644
--- a/Directory.Build.targets
+++ b/Directory.Build.targets
@@ -5,6 +5,7 @@
   <Import Project="$(RepositoryEngineeringDir)packageContent.targets" />
   <Import Project="$(RepositoryEngineeringDir)versioning.targets" />
   <Import Project="$(RepositoryEngineeringDir)ApiCompatibility\PublicApiAnalyzer.targets" />
+  <Import Project="$(RepositoryEngineeringDir)Tests.targets" Condition="'$(IsTestProject)' == 'true'" />
 
   <!-- CA1416 Validate platform compatibility. Do not apply to netstandard as it does not have the attribute. -->
   <ItemGroup>
```


There're bunch of MSBuild definitions with `Condition="'$(IsTestProject)' == 'true'"` scatteredd throughout which should be moved into Tests.props and Tests.targets.

Common test-related package references should as well be moved there - this removes the need to modify individual test projects.
E.g., all of this is a maintenance overhead:
image

File: eng/pipelines/build.yml [hunk 3]
```diff
@@ -101,6 +101,32 @@ jobs:
         /m:1
       displayName: Run Unit Tests
       condition: and(eq(variables['System.TeamProject'], 'public'), ne('${{ parameters.targetArchitecture }}', 'arm64'))
+    
+    # Code Coverage
+    - script: dotnet tool install --global dotnet-coverage
+      displayName: Install dotnet-coverage
+      condition: and(eq(variables['_BuildConfig'], 'Debug'), eq('${{ parameters.targetArchitecture }}', 'x64'))
```
Comment (RussKie) [inline eng/pipelines/build.yml:108]: Instead of repeating the condition you could place all of the tasks under a single condition

```yaml
- ${{ if and(eq(variables['_BuildConfig'], 'Debug'), eq('${{ parameters.targetArchitecture }}', 'x64')) }}:
  - script: dotnet tool install --global dotnet-coverage
    displayName: Install dotnet-coverage

  - script: ...

  - script: ...

```

Comment (SimonZhao888) [reply eng/pipelines/build.yml]: Great advice, let me know if you have any suggestions and I'll tweak them later!

File: eng/pipelines/build.yml [hunk 4]
```diff
@@ -134,14 +134,29 @@ jobs:
...
+
     # Upload code coverage data
     - script: $(Build.SourcesDirectory)/.dotnet/dotnet msbuild -restore
         eng/CodeCoverage.proj
         /p:Configuration=$(_BuildConfig)
         $(_InternalRuntimeDownloadArgs)
         /bl:$(BUILD.SOURCESDIRECTORY)\artifacts\log\$(_BuildConfig)\CodeCoverage-${{ parameters.targetArchitecture }}.binlog
       displayName: Upload coverage to codecov.io
-      condition: and(succeeded(), eq(variables._Coverage, 'true'))
+      condition: and(succeeded(), eq(variables._Coverage, 'true'), eq('${{ parameters.targetArchitecture }}', 'x64'))
```
Comment (SimonZhao888) [inline eng/pipelines/build.yml:159]: Hi @RussKie , 
Could we use this one task to enable code coverage testing? 
Except the PublishCodeCoverageResults@1 task, we can see that it also performs the merge and upload coverage data tasks, can we try to modify it to do this work?

https://github.com/dotnet/winforms/blob/9e2fc95fca4e6d18444c819b35bb1338ac6ca63d/eng/CodeCoverage.proj#L8-L11

File: eng/pipelines/build.yml [hunk 5]
```diff
@@ -134,14 +134,29 @@ jobs:
...
+        reportDirectory: '$(Build.SourcesDirectory)\artifacts\CoverageResult\$(_BuildConfig)'
+      displayName: PublishCodeCoverageResults
+      condition: and(eq(variables['_BuildConfig'], 'Debug'), eq('${{ parameters.targetArchitecture }}', 'x64'))
+
     # Upload code coverage data
     - script: $(Build.SourcesDirectory)/.dotnet/dotnet msbuild -restore
         eng/CodeCoverage.proj
         /p:Configuration=$(_BuildConfig)
         $(_InternalRuntimeDownloadArgs)
         /bl:$(BUILD.SOURCESDIRECTORY)\artifacts\log\$(_BuildConfig)\CodeCoverage-${{ parameters.targetArchitecture }}.binlog
```
Comment (SimonZhao888) [inline eng/pipelines/build.yml:157]: Hi @RussKie, 
Could we use this task to enable code coverage testing? 
Except the PublishCodeCoverageResults@1 task, we can see that it also performs the merge and upload coverage data tasks, can we try to modify it to do this work?

https://github.com/dotnet/winforms/blob/9e2fc95fca4e6d18444c819b35bb1338ac6ca63d/eng/CodeCoverage.proj#L8-L11

Comment (RussKie) [reply eng/pipelines/build.yml:157]: Probably... Though since I don't use codecov, I won't be able to provide with any meaningful help. This is probably more a @sharwell type of question.

File: eng/pipelines/build.yml (hunk 3 above)
Comment (SimonZhao888) [reply eng/pipelines/build.yml]: It's hard to place all those tasks under a single condition, since these two parameters are not assigned values when pipeline is built.

File: eng/MergeCoverageData.ps1 [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+dotnet tool install dotnet-reportgenerator-globaltool --tool-path $converageToolsPath
+
+$converageTools = Join-Path $converageToolsPath "reportgenerator.exe"
+Test-Path -Path $converageTools
+$reportCoverage = Join-Path $artifactsDir "bin\*\Debug\*\coverage\*.coverage"
+$resultDir = Join-Path $artifactsDir "CoverageResult\Debug"
+$reporttype = 'Html;HtmlInline_AzurePipelines'
+$reporttypeXml = 'Cobertura'
+$filter = '+System.Windows.Forms*;-*Tests.dll'
```
Comment (LeafShi1) [inline eng/MergeCoverageData.ps1:24]: Can the filtering be more detailed? The results show that some files do not exist
image

File: eng/MergeCoverageData.ps1 [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+if (!(Test-Path -Path $converageToolsPath)) {
+  New-Item -Path $converageToolsPath -ItemType Directory
+}
+dotnet tool install dotnet-reportgenerator-globaltool --tool-path $converageToolsPath
+
+$converageTools = Join-Path $converageToolsPath "reportgenerator.exe"
+Test-Path -Path $converageTools
+$reportCoverage = Join-Path $artifactsDir "bin\*\Debug\*\coverage\*.coverage"
+$resultDir = Join-Path $artifactsDir "CoverageResult\Debug"
+$reporttype = 'Html;HtmlInline_AzurePipelines'
```
Comment (LeafShi1) [inline eng/MergeCoverageData.ps1:22]: Only one report type is needed here, right?

File: eng/pipelines/build.yml (hunk 5 above)
Comment (SimonZhao888) [reply eng/pipelines/build.yml]: This task also doesn't have any effect on the upload code coverage results, and in the latest commit I tried to log out of this task. coverage results can still be displayed. here is the pipeline link: https://dev.azure.com/dnceng-public/public/_build/results?buildId=509880&view=codecoverage-tab

File: eng/MergeCoverageData.ps1 (hunk 7 above)
Comment (SimonZhao888) [reply eng/MergeCoverageData.ps1]: Done.

File: eng/MergeCoverageData.ps1 (hunk 6 above)
Comment (SimonZhao888) [reply eng/MergeCoverageData.ps1]: Done.
image

File: eng/pipelines/build.yml (hunk 5 above)
Comment (sharwell) [reply eng/pipelines/build.yml]: > The PublishCodeCoverageResults@1 task will publish the coverage results.

This task does not publish code coverage results in a usable form.

File: eng/pipelines/build.yml [hunk 8]
```diff
@@ -134,14 +134,20 @@ jobs:
...
-        /bl:$(BUILD.SOURCESDIRECTORY)\artifacts\log\$(_BuildConfig)\CodeCoverage-${{ parameters.targetArchitecture }}.binlog
-      displayName: Upload coverage to codecov.io
-      condition: and(succeeded(), eq(variables._Coverage, 'true'))
+    # Merge code coverage data
+    - pwsh: eng\MergeCoverageData.ps1 -configuration $(_BuildConfig) -artifactsDir '$(Build.SourcesDirectory)\artifacts'
+      displayName: Merge Coverage Data
+      condition: and(eq(variables['_BuildConfig'], 'Debug'), eq('${{ parameters.targetArchitecture }}', 'x64'))
+
+    #Publish code coverage results
+    - task: PublishCodeCoverageResults@1
```
Comment (sharwell) [inline eng/pipelines/build.yml:143]: ❓ Why did this switch from codecov.io (a service which provides meaningful code coverage data) to Azure (a service which only provides the meaningless portion of code coverage data)?

**Comment by sharwell** [comment]
The removal of codecov.io upload is highly unfortunate.

**Comment by SimonZhao888** [comment]
> but if the data only goes to one location it should be codecov.io.

OK, I've added this task back.

File: eng/MergeCoverageData.ps1 [hunk 9]
```diff
@@ -0,0 +1,26 @@
+<#
+.SYNOPSIS
+Downloads 32-bit and 64-bit procdump executables and returns the path to where they were installed.
```
Comment (Tanya-Solyanik) [inline eng/MergeCoverageData.ps1:3]: Where are we downloading it from? I don't see any references to procdump below....

**Comment by Tanya-Solyanik** [comment]
@SimonZhao888  - Would it make sense to split this PR into 2 PRs or 2 commits. First do a cleanup and refactoring tasks, and then  enable the code coverage? It will be easier to review and maintain this way.  This is what Sam did with his big PR - https://github.com/dotnet/winforms/pull/10552

File: eng/pipelines/build.yml [hunk 10]
```diff
@@ -134,14 +134,29 @@ jobs:
         /bl:$(BUILD.SOURCESDIRECTORY)\artifacts\log\$(_BuildConfig)\PackSignPublish-${{ parameters.targetArchitecture }}.binlog
       displayName: Pack, Sign, and Publish
 
+    # Merge code coverage data
+    - pwsh: eng\MergeCoverageData.ps1 -configuration $(_BuildConfig) -artifactsDir '$(Build.SourcesDirectory)\artifacts'
+      displayName: Merge Coverage Data
+      condition: and(eq(variables['_BuildConfig'], 'Debug'), eq('${{ parameters.targetArchitecture }}', 'x64'))
+
+    #Publish code coverage results
+    - task: PublishCodeCoverageResults@1
```
Comment (Tanya-Solyanik) [inline eng/pipelines/build.yml:143]: We have PR and CI builds in GH, and in the internal AzDO repo. Where is this task executed?

File: eng/pipelines/build.yml (hunk 5 above)
Comment (Tanya-Solyanik) [reply eng/pipelines/build.yml:157]: Do we need this task? Can we remove it if it's not useful? Or does it have any side-effects?

**Comment by sharwell** [comment]
I may be able to help with it as well once the integration test changes are completed.

File: eng/pipelines/build.yml (hunk 5 above)
Comment (sharwell) [reply eng/pipelines/build.yml:157]: This task is the one that uploads to codecov.io, so it's arguably the most important task in the feature.

Comment (Tanya-Solyanik) [reply eng/pipelines/build.yml:157]: I misunderstood your comment then.

Comment (sharwell) [reply eng/pipelines/build.yml:157]: @Tanya-Solyanik I was referring to `PublishCodeCoverageResults@1` as the problematic task. We use a script instead.

File: eng/MergeCoverageData.ps1 (hunk 9 above)
Comment (SimonZhao888) [reply eng/MergeCoverageData.ps1]: I've removed this.

File: eng/pipelines/build.yml (hunk 10 above)
Comment (SimonZhao888) [reply eng/pipelines/build.yml:143]: Since [PublishCodeCoverageResults@1](https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/publish-code-coverage-results-v1?view=azure-pipelines#code-coverage-results-for-javascript-with-istanbul-using-yaml) is an AzDO  internal task, I assume it will be executed at AzDO.

Comment (sharwell) [reply eng/pipelines/build.yml]: I can't tell if this is asking about GitHub Actions vs Azure Pipelines, or if it's asking about Azure Pipelines public build vs Azure Pipelines internal/signed build.

Comment (SimonZhao888) [reply eng/pipelines/build.yml]: Hi @sharwell, 

Could you please help me with a question?
Now we are going to count the code coverage of all the automated tests, I am checking the coverage results on CodeCov server, but found that the results counted on CodeCov server are much different from the results on Azure pipeline, I am not sure if I am missing any settings, Could you please point them out for me?

Pipeline link: [codecoverage-tab](https://dev.azure.com/dnceng-public/public/_build/results?buildId=515745&view=codecoverage-tab)

CodeCov link:  [CodeCov-codecoverage](https://app.codecov.io/gh/dotnet/winforms/tree/SimonZhao888%2Fv-weidzh-winforms%3Adev%2Fv-weidzh%2FUnitTestCoverage/src?displayType=tree)
image

Comment (sharwell) [reply eng/pipelines/build.yml]: > Now we are going to count the code coverage of all the automated tests

Yes, this is hugely beneficial in identifying certain categories of accidental bugs and/or behavior changes.

>  I am checking the coverage results on CodeCov server, but found that the results counted on CodeCov server are much different from the results on Azure pipeline

I noticed the same. I was planning to make some tweaks either after you are ready for final review/merge or after it merges. Once you reach a stable point I can go over those details with you.

Comment (sharwell) [reply eng/pipelines/build.yml]: Items I'm planning to verify later (could be part of this PR or separate):

* Ensure we only upload exactly one coverage report to codecov.io for a single PR submission. Historically (talking 2+ years ago), codecov.io internally was not efficient at merging multiple reports, which could cause problems for large repositories. I found that merging the reports before submitting them significantly improved response times and reliability.

I guess I only have the one item right now. During my work for this item, I'll be comparing the behavior of WinForms with the behavior of dotnet/roslyn-analyzers (uses coverlet+Arcade which we want to match) and DotNetAnalyzers/StyleCopAnalyzers (uses OpenCover but has a proper [Publish Code Coverage stage](https://dev.azure.com/DotNetAnalyzers/StyleCopAnalyzers/_build/results?buildId=6054&view=results) that I can model after if we need it).

File: codecov.yml [hunk 11]
```diff
@@ -20,16 +20,28 @@ flags:                            # which files to include in the reporting
...
       - src/System.Drawing/src/
+      - src/System.Drawing.Common/src/
       - src/System.Drawing.Design/src/
       - src/System.Windows.Forms/src/
+      - src/System.Windows.Forms.Analyzers/src/
+      - src/System.Windows.Forms.Analyzers.CSharp/src/
       - src/System.Windows.Forms.Design/src/
       - src/System.Windows.Forms.Design.Editors/src/
+      - src/System.Windows.Forms.Primitives/src/
+      - src/System.Windows.Forms.PrivateSourceGenerators/src/
```
Comment (Tanya-Solyanik) [inline codecov.yml:35]: Please add a new assembly - System.Private.Windows.Core

File: eng/CodeCoverage.proj [hunk 12]
```diff
@@ -23,8 +23,8 @@
 
     <!-- Merge multiple coverlet reports into a single Cobertura report before uploading to codecov.io, in order to
       reduce upload size and load on the codecov.io processing servers. -->
-    <Message Importance="high" Text="&quot;$(_ReportGeneratorPath)&quot; &quot;-reports:@(_CoverageReports)&quot; -targetdir:$(BaseOutputPath)coverage -reporttypes:Cobertura" />
-    <Exec Command="&quot;$(_ReportGeneratorPath)&quot; &quot;-reports:@(_CoverageReports)&quot; -targetdir:$(BaseOutputPath)coverage -reporttypes:Cobertura" />
+    <Message Importance="high" Text="&quot;$(_ReportGeneratorPath)&quot; &quot;-reports:@(_CoverageReports)&quot; -targetdir:$(BaseOutputPath)coverage -reporttypes:Cobertura  -filefilters:-*.g.cs" />
```
Comment (Tanya-Solyanik) [inline eng/CodeCoverage.proj:26]:
```suggestion
    <Message Importance="high" Text="&quot;$(_ReportGeneratorPath)&quot; &quot;-reports:@(_CoverageReports)&quot; -targetdir:$(BaseOutputPath)coverage -reporttypes:Cobertura -filefilters:-*.g.cs" />
```

File: eng/CodeCoverage.proj [hunk 13]
```diff
@@ -23,8 +23,8 @@
...
+    <Exec Command="&quot;$(_ReportGeneratorPath)&quot; &quot;-reports:@(_CoverageReports)&quot; -targetdir:$(BaseOutputPath)coverage -reporttypes:Cobertura  -filefilters:-*.g.cs" />
```
Comment (Tanya-Solyanik) [inline eng/CodeCoverage.proj:27]:
```suggestion
    <Exec Command="&quot;$(_ReportGeneratorPath)&quot; &quot;-reports:@(_CoverageReports)&quot; -targetdir:$(BaseOutputPath)coverage -reporttypes:Cobertura -filefilters:-*.g.cs" />
```

File: eng/pipelines/build.yml [hunk 14]
```diff
@@ -133,15 +133,25 @@ jobs:
         $(_InternalRuntimeDownloadArgs)
         /bl:$(BUILD.SOURCESDIRECTORY)\artifacts\log\$(_BuildConfig)\PackSignPublish-${{ parameters.targetArchitecture }}.binlog
       displayName: Pack, Sign, and Publish
-
+
```
Comment (Tanya-Solyanik) [inline eng/pipelines/build.yml:136]: Please revert this change

File: src/System.Windows.Forms.Primitives/tests/TestUtilities.Tests/System.Windows.Forms.Primitives.TestUtilities.Tests.csproj [hunk 15]
```diff
@@ -1,4 +1,4 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms.Primitives/tests/TestUtilities.Tests/System.Windows.Forms.Primitives.TestUtilities.Tests.csproj:1]: Please revert this change

**Comment by Tanya-Solyanik** [comment]
Will a diff between coverage before the PR and coverage after the PR be published when PR build completes?

**Comment by sharwell** [comment]
> Will a diff between coverage before the PR and coverage after the PR be published when PR build completes?

That is the expectation. It's not clear yet why it's not showing up in this repository, but you can see it in action here:
https://github.com/dotnet/roslyn-analyzers/pull/7116

At the time of this comment, it says there are 38 lines changed in my linked PR which are not fully covered, out of a total of around 400 LOC touched by the PR.

**Comment by SimonZhao888** [comment]
> That is the expectation. It's not clear yet why it's not showing up in this repository, but you can see it in action here:
> [dotnet/roslyn-analyzers#7116](https://github.com/dotnet/roslyn-analyzers/pull/7116)

I'll double-check the differences in these two pipelines to enable the feature in codecov.io.

File: Directory.Build.props [hunk 16]
```diff
@@ -70,7 +70,7 @@
     <SingleHit>true</SingleHit>
     <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
     <CoverletOutputFormat>opencover</CoverletOutputFormat>
-    <Include>[System.Windows.*]*</Include>
+    <Include>[System.Windows.*]*,[Microsoft.VisualBasic.*]*</Include>
```
Comment (Tanya-Solyanik) [inline Directory.Build.props:73]: Should you include System.Drawing.Common and System.Private?

**Review by Tanya-Solyanik — APPROVED** [review]
Thank you! Make sure to squash-merge this PR!

**Comment by Tanya-Solyanik** [comment]
> > That is the expectation. It's not clear yet why it's not showing up in this repository, but you can see it in action here:
> > [dotnet/roslyn-analyzers#7116](https://github.com/dotnet/roslyn-analyzers/pull/7116)
> 
> I'll double-check the differences in these two pipelines to enable the feature in codecov.io.

Feel free to do that in the next PR.

**Review by JeremyKuhne — APPROVED** [review]

**Review by lonitra — APPROVED** [review]

File: codecov.yml [hunk 17]
```diff
@@ -20,16 +20,29 @@ flags:                            # which files to include in the reporting
   production:
     paths:
       - src/Common/src/
+      - src/Microsoft.VisualBasic/src/
+      - src/Microsoft.VisualBasic.Forms/src/
       - src/System.Design/src/
       - src/System.Drawing/src/
```
Comment (Tanya-Solyanik) [inline codecov.yml:26]: We don't have to include System.Design into reporting, do we still need it here?

**Review by Tanya-Solyanik — APPROVED** [review]

File: codecov.yml (hunk 17 above)
Comment (SimonZhao888) [reply codecov.yml:26]: The changes here are mainly for Codecov.io, if, however, they are not needed, I will remove them in the PR adapted to Codecov.io.
